src/test/java/net/datafaker/providers/base/TimeAndDateTest.java

```diff
@@ -0,0 +1,229 @@
+package net.datafaker.providers.base;
+
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TimeAndDateTest extends BaseFakerTest<BaseFaker> {
+
+    private final TimeAndDate timeAndDate = faker.timeAndDate();
+
+    @RepeatedTest(100)
+    void testFutureDate() {
+        assertThat(timeAndDate.future()).isInTheFuture();
+    }
+
+    @RepeatedTest(1000)
+    void testFutureDateWithBounds() {
+        Instant now = Instant.now();
+        Instant future = timeAndDate.future(1, TimeUnit.SECONDS, now);
+        assertThat(future).isBetween(now, now.plusSeconds(1));
+    }
+
+    @RepeatedTest(1000)
+    void testFutureDateWithMinimum() {
+        Instant now = Instant.now();
+        Instant future = timeAndDate.future(5, 4, TimeUnit.SECONDS);
+        assertThat(future)
+            .isBetween(now.plusMillis(3500), now.plusMillis(5500));
```

Comment:
Make constants for these magic numbers or add a comment that clarifies. I'm sure it makes sense if I read the code but less digging is better 😉

Reply:
I agree, but I mostly ported the old code to the new types. I'll have a look later.

Reply:
This is actually great feedback. I'm reading this code now, and to me, it doesn't really make sense. Why are 5 and 4 flipped around? It would make more sense to me to have a future date between 4 and 5 seconds into the future, not between 5 and 4. What do you think? (the challenge is that the behaviour of this method would be slightly different from the old one, so maybe this should throw an exception if argument 2 is lower than argument 1? Thoughts?

Reply:
Hahaha I was actually on about the 3500 & 5500. 

But ya I agree changing the item you're talking about seems better too and enforcing the condition as well.